pkg/backup/backup.go

```diff
@@ -645,6 +645,17 @@ func (kb *kubernetesBackupper) executeItemBlockActions(
 				name:          relatedItem.Name,
 				inItemBlock:   true,
 			})
+
+			relatedItemMetadata, err := meta.Accessor(item)
+			if err != nil {
+				log.WithError(errors.WithStack(err)).Warn("Failed to get object metadata.")
+				continue
+			}
+			// Don't add to ItemBlock if item is excluded
+			// itemInclusionChecks logs the reason
+			if !itemBlock.itemBackupper.itemInclusionChecks(log, false, relatedItemMetadata, item, relatedItem.GroupResource) {
+				continue
```

Comment:
```diff
			itemsMap[relatedItem] = append(itemsMap[relatedItem], &kubernetesResource{
				groupResource: relatedItem.GroupResource,
				preferredGVR:  gvr,
				namespace:     relatedItem.Namespace,
				name:          relatedItem.Name,
-				inItemBlock:   true,
+				inItemBlock:   false,
			})
```
Shouldn't this code in line 646 be flipped to false?

Reply:
@kaovilai No. "True" means we've already processed this for item block exclusion -- it prevents Velero from adding it to another ItemBlock later. Since we already know we don't want to include this in an ItemBlock, we set it to true here so we don't have to even consider it later.

I guess `processedForItemBlock` would have been a more accurate field name here, but that's clumsy and might also be confusing. I think the current name is probably clearest, but maybe a comment here explaining why it's set to true would be appropriate?

Reply:
or just `processed` indicating item was processed. a comment to clarify that inItemBlock does not mean it would be in an item block would be great.

Reply:
@kaovilai What about `inItemBlockOrExcluded`? That covers the default use case of "this has already been included" more explicitly than "processed" without being confusing in the exclusion case?

Reply:
works for me.